receiver/elasticapmreceiver/internal/mappers/intakeV2ToOtlpTopLevelFields.go

```diff
@@ -0,0 +1,116 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package mappers
+
+import (
+	"encoding/hex"
+	"strings"
+	"time"
+
+	"github.com/elastic/apm-data/model/modelpb"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.uber.org/zap"
+)
+
+type TopLevelFieldSetter interface {
+	SetTraceID(v pcommon.TraceID)
+	SetSpanID(v pcommon.SpanID)
+}
+
+// Shared across LogRecord and Span
+func SetTopLevelFieldsCommon(event *modelpb.APMEvent, timestamp time.Time, t TopLevelFieldSetter, logger *zap.Logger) {
+
+	if event.Trace != nil && event.Trace.Id != "" {
+		traceId, err := traceIDFromHex(event.Trace.Id)
+		if err == nil {
+			t.SetTraceID(traceId)
+		} else {
+			logger.Error("failed to parse trace ID", zap.String("trace_id", event.Trace.Id))
+		}
```

Comment:
Should we continue processing the event in case of error? Should it be a `Warn` or lower log level?

In case of being an error, wdyt of fail fast and remove the logger parameter?


```suggestion
		traceId, err := traceIDFromHex(event.Trace.Id)
		if err != nil {
		     return fmt.Errorf("failed to parse trace ID: %w", err)
		} 
		t.SetTraceID(traceId)
```

Reply:
I think this was discussed in a previous PR. So there is no enforcement for these ids after this point anywhere. So if parsing fails, there will be still error, transaction, and span documents stored in elasticsearch with all the remaining fields. Of course the value of such documents won't be very useful - correlating events won't work. That's why we went with an error level log.

On the other hand, I'd not stop the processing. Reason we ended up with this is that if we continue here, there will be still documents in elasticsearch - other fields will be set, plus we print an error. This is probably a better experience for users than potentially not storing anything in elasticsearch.